x-pack/plugins/apm/public/components/app/TransactionDetails/Transaction/WaterfallContainer/Waterfall/TransactionFlyout/index.tsx

```diff
@@ -104,6 +104,8 @@ export function TransactionFlyout({
     return null;
   }
 
+  const waterfallItem = waterfall.itemsById[transactionDoc.transaction.id];
+
```

Comment:
We already do this in:
https://github.com/elastic/kibana/blob/83a2f7b0307f7fa6bccb5681a5462a071e0ed937/x-pack/plugins/apm/public/components/app/TransactionDetails/Transaction/WaterfallContainer/Waterfall/index.tsx#L86-L88

I think `TransactionFlyout` should not take `waterfall` as an argument. It should be consistent with `SpanFlyout`  which also doesn't take `waterfall`. Instead `SpanFlyout` take `totalDuration={waterfall.duration}`.

Similarly `TransactionFlyout` should take 
```
traceRootDuration={waterfall.traceRootDuration}
errorCount={currentItem.errorCount}
```

I think that will simplify some things.